src/basic/time-util.c

```diff
@@ -357,6 +359,14 @@ char *format_timestamp_style(
                 return buf;
         }
 
+        if (style == TIMESTAMP_UNIX) {
+                r = snprintf(buf, l, "@" USEC_FMT, t / USEC_PER_SEC);  /* round down µs → s */
+                if (r < 0 || (size_t) r >= l)
+                        return NULL; /* Doesn't fit */
+
+                return buf;
+        }
```

Comment:
i think this should be moved 6 lines up, i.e. before the > 9999 thing. Because the the check it does doesn't apply to TIMESTAMP_UNIX at all: we can output anything we can store in usec_t pretty much.

Reply:
Updated!